### ISSUE TITLE
Man in the Middle

### ISSUE THREAD

**Issue by Gunni** [ampproject/amphtml#6144]
## Short description of your issue:

TL;DR; My issue is not with AMP, my issue is with the AMP CDN, which i do not want intercepting my requests!

When saved to inbox using the inbox save to inbox feature, and then opening on another computer, the cdn.ampproject.org is placed before my url, intercepting the request and proxying the response to me.

I consider this an attack, the amp cdn is not only receiving information about where i browse, but it also proxies the response and is effectively a MITM proxy with all the issues those have!

The CDN could put _anything_ in the webpages i view, ads, spying cookies, viruses, anything! And they also become a jucy target for hackers that want to affect a lot of users.

## How do we reproduce the issue?

1. https://www.reddit.com/r/talesfromtechsupport/comments/5cb0ju/encyclop%C3%A6dia_moronica_y_is_for_you_can_is_not_you/
2. Save to Inbox using the inbox app for chrome
3. Go to https://inbox.google.com/
4. Click the saved item
5. Click View Link
6. End up on https://cdn.ampproject.org/c/s/amp.reddit.com/r/talesfromtechsupport/comments/5cb0ju/encyclop%C3%A6dia_moronica_y_is_for_you_can_is_not_you/
7. Observe that link is not the saved link

Why not https://www.reddit.com/r/talesfromtechsupport/comments/5cb0ju/encyclop%C3%A6dia_moronica_y_is_for_you_can_is_not_you/

Or at the very least https://amp.reddit.com/r/talesfromtechsupport/comments/5cb0ju/encyclop%C3%A6dia_moronica_y_is_for_you_can_is_not_you/ ?

## What browsers are affected?

Chrome

**Comment by jridgewell** [comment]
/to @rudygalfi. Or maybe @cramforce would like to correct this.

**Comment by lannka** [comment]
For your particular inbox related problem, the fix might be to have the inbox Chrome extension to read the canonical URL from the AMP page meta data, instead of using the CDN URL.

Other than that, I don't see it any kind of attack, because you're most likely on Google.com, and amp CDN is just another CDN service that Google is using to cache pages that publishers agreed to be cached.

**Comment by cramforce** [comment]
Correct. Since the CDNs URLs are created by platforms that already have control over where user is going, this is a classic "You can get root when you already have root" situation.

The AMP Cache is stateless, so it cannot really have any user specific information on it.

It is correct that AMP Caches are an interesting target. Providers of AMP Caches have to take extra care, similar to large ISPs like Comcast.
